File: QViz.NET/Periscope.QViz/JSON/Role.cs
using System;
using System.Collections.Generic;

namespace Periscope.QViz.JSON
{
	/// <summary>
	/// JSON Model for User Roles
	/// </summary>
	public class Role
	{
		/// <summary>
		/// Identifier of the User Role
		/// </summary>
		public string roleId;
		
		/// <summary>
		/// Name of the User Role
		/// </summary>
		public string name;
		
		/// <summary>
		/// Is this User Role Active?
		/// </summary>
		public bool isActive;
		
		/// <summary>
		/// Date and Time on which this Model was Created
		/// </summary>
		public DateTime createdOn;
		
		/// <summary>
		/// Date and Time on which this Model was last Updated
		/// </summary>
		public DateTime updatedOn;
		
		/// <summary>
		/// Identifier of the Company that the User Role belongs to
		/// </summary>
		public string companyId;
		
		/// <summary>
		/// List of User Profile Objects associated with the User Role
		/// </summary>
		public IList<UserProfile> userProfiles;
		
		/// <summary>
		/// List of Role Permission Objects associated with the User Role
		/// </summary>
		public IList<RolePermission> rolePermissions;

		/// <summary>
		/// Default Constructor
		/// </summary>
		public Role()
		{
			this.roleId = "";
			this.name = "";
			this.isActive = false;
			this.createdOn = DateTime.Now;
			this.updatedOn = DateTime.Now;
			this.companyId = "";
			this.userProfiles = new List<UserProfile>();
			this.rolePermissions = new List<RolePermission>();
		}
	}
}

File: QViz.Java/src/main/java/com/periscope/qviz/json/Role.java
package com.periscope.qviz.json;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * JSON Model for User Roles
 */
public class Role {

	/**
	 * Identifier of the User Role
	 */
	public String roleId;

	/**
	 * Name of the User Role
	 */
	public String name;

	/**
	 * Is this User Role Active?
	 */
	public Boolean isActive;

	/**
	 * Date and Time on which this Model was Created
	 */
	public Date createdOn;

	/**
	 * Date and Time on which this Model was last Updated
	 */
	public Date updatedOn;

	/**
	 * Identifier of the Company that the User Role belongs to
	 */
	public String companyId;

	/**
	 * List of User Profile Objects associated with the User Role
	 */
	public List<UserProfile> userProfiles;

	/**
	 * List of Role Permission Objects associated with the User Role
	 */
	public List<RolePermission> rolePermissions;

	/**
	 * Default Constructor
	 */
	public Role() {
		this.roleId = "";
		this.name = "";
		this.isActive = false;
		this.createdOn = null;
		this.updatedOn = null;
		this.companyId = "";
		this.userProfiles = new ArrayList<>();
		this.rolePermissions = new ArrayList<>();
	}

}
